src/index.py

```diff
@@ -1,82 +1,53 @@
 import logging
 import os
-from typing import Dict, List, Optional
 
-import yaml
-from cognite.experimental import CogniteClient
-from cognite.experimental.data_classes import Function
-
-from config import FunctionConfig, TenantConfig
-from function import deploy_function
+from config import FunctionConfig, TenantConfig, create_experimental_cognite_client
+from function import try_delete, upload_and_create
+from github_log_handler import GitHubLogHandler
 from schedule import deploy_schedule
 
+# Configure logging:
+root_logger = logging.getLogger()
+root_logger.setLevel(logging.INFO)
+root_logger.addHandler(GitHubLogHandler())
 
-def main(config: FunctionConfig) -> Optional[Function]:
-    client = CogniteClient(
-        api_key=config.tenant.deployment_key,
-        project=config.tenant.cdf_project,
-        base_url=config.tenant.cdf_base_url,
-        client_name="function-action",
-    )
-    f = deploy_function(client, config)
-
-    if f is not None:
-        deploy_schedule(client, f, config)
-
-    return f
+logger = logging.getLogger(__name__)
 
 
-class GitHubLogHandler(logging.StreamHandler):
-    def __init__(self, stream=None):
-        super(GitHubLogHandler, self).__init__(stream=stream)
+def main(config: FunctionConfig) -> None:
+    client = create_experimental_cognite_client(config.tenant)
 
-    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-a-debug-message
-    def format(self, record):
-        message = super(GitHubLogHandler, self).format(record)
-        level_map: Dict = {
-            logging.CRITICAL: "error",
-            logging.ERROR: "error",
-            logging.WARNING: "warning",
-            logging.INFO: "warning",
-            logging.DEBUG: "debug",
-            logging.NOTSET: "warning",
-        }
-        return (
-            f"::{level_map.get(record.levelno)} file={record.filename},line={record.levelno}::{record.name}: {message}"
-        )
+    # Delete old function and file:
+    try_delete(client, config.external_id)
+    if config.remove_only:
+        return
```

Comment:
Rewrote the remove-only logic so that I could remove an entire middle-layer function